packages/commons/src/typeUtils.ts

```diff
@@ -0,0 +1,177 @@
+/**
+ * Returns true if the passed value is a record (object).
+ *
+ * @param value The value to check
+ */
+const isRecord = (value: unknown): value is Record<string, unknown> => {
```

Comment:
the keys in the `Record` doesn't have to strings only, afaik we can have `string | number | symbol`